packages/backend/src/core/preprocessing/UserStatisticsPreprocessor.ts

```diff
@@ -117,6 +116,51 @@ export class UserStatisticsPreprocessor {
     }
   }
 
+  async catchUpL2Transactions(
+    trx: Knex.Transaction,
+    preprocessToStateUpdateId: number
+  ) {
+    const recordsToUpdate =
+      await this.preprocessedUserStatisticsRepository.getAllWithoutL2TransactionStatisticsUpToStateUpdateId(
```

Comment:
I'm afraid that this approach is incorrect. What it does is it gets state update statistics for each user, and then updates them with l2 statistics for those users. 

But what if a user buys 1 ETH and then sells 1 ETH in the same state update. There will be no entry for this user in the state update (their balance didn't change), even though he made 2 transactions. With current implementation it will miss these 2 transactions completely. 

It seems that proper implementation should go straight to l2_transactions table for given state update, fetch unique set of users (starkKeys) and then calculate statistics for each user. Those statisics should be written to existing statistics record, or a new record needs to be added with zero's in columns for state update statistics.

So either implement as suggested above, or let's have a separate table with only l2 preprocessed statistics... Let me know what you prefer.